### DISCUSSION TITLE
possible to access `vfile.data` when using `@mdx-js/esbuild`?

### DISCUSSION THREAD

**Discussion by stefanprobst** [mdx-js/mdx#2055] [Q&A] [answered]
is it possible to somehow access any `vfile.data` generated by plugins when using `@mdx-js/esbuild`?

**Comment by wooorm** [comment, marked as answer]
a) it probably helps to know what you want to do and why, as there are perhaps alternative solutions to your root problem
b) How would you want to access it on the esbuild side? Does esbuild have an arbitrary data store?

**Reply by stefanprobst** [reply]
some background:

i've been playing with [`contentlayer`](https://www.contentlayer.dev) which uses a `unified` pipeline for markdown processing, and `mdx-bundler` for mdx processing. i wanted to make `vfile.data` from the markdown piipeline available [in this PR](https://github.com/contentlayerdev/contentlayer/pull/236) and was thinking about how to do the same with mdx. i haven't used `mdx-bundler` but from what i understand it basically wraps `@mdx-js/esbuild` and it was not clear to me if it was somehow possible to return `vfile.data` from [here](https://github.com/mdx-js/mdx/blob/main/packages/esbuild/lib/index.js#L226-L233) - but maybe this is more of a question about `esbuild` itself?

**Reply by wooorm** [reply]
If you want to expose stuff as an export inside a file, like I see in your PR, that could be a plugin. It wouldn’t be related to contentlayer or mdx-bundler but directly hook into MDX, like the other MDX plugins: https://mdxjs.com/docs/extending-mdx/#list-of-plugins.

If you want to expose stuff to esbuild, it’s first a question whether esbuild has an API for that. That depends on esbuild. And then we could indeed pass info back, in the lines you linked!
